### ISSUE TITLE
The first two characters cannot be deleted when deleting multiple lines

### ISSUE THREAD

**Issue by Dreamhai** [xtermjs/xterm.js#5138]
## Details
- Browser and browser version: Chrome  version: 127.0.6533.72
- OS version: macOS  14.2
- xterm.js version: 5.5.0

### Steps to reproduce

https://xtermjs.org/

The demo on the official website cannot be deleted after entering multiple lines of characters and deleting the first two characters
![Uploading 截屏2024-08-27 14.53.47.png…]()

**Comment by jerch** [comment]
Thats an issue of the line edit shim here: https://github.com/jerch/xtermjs.org/blob/c35fa212cc458b93d4d3505b0ca4e27d047f10fd/js/demo.js#L31

Note that the shim there never was intended to give a full featured experience, in fact writing a proper line editor for the terminal is quite involved. A real shell prompt would handle that correctly with its line edit lib like libreadline.

**Comment by Dreamhai** [comment]
if I remove "if (term._core.buffer.x > 2) {}"，Can only delete the current line.

how to delete to the previous line? The cursor stays at the first position of the second line and does not jump back to the last character position of the first line

**Comment by jerch** [comment]
> how to delete to the previous line? The cursor stays at the first position of the second line and does not jump back to the last character position of the first line

Yes, by default the cursor is bound to a single row in a terminal for simple control chars (thats inherited from typewriters, which also cannot jump back to a higher line). What a line edit lib typically does here - buffer the (multiline) prompt content, and redraw the full content with the help of cursor move terminal sequences.

**Comment by Tyriar** [comment]
Closing this as designed as the scope of the demo is limited and spending too much time there would be a waste.
